GameServerLib/Logic/GameObjects/AttackableUnits/AI/Minion.cs

```diff
@@ -97,52 +95,65 @@ public override void Update(float diff)
 
                 if (ScanForTargets()) // returns true if we have a target
                 {
-                    KeepFocussingTarget(); // fight target
+                    //Here we need to add another check: Positioning.
+                    if (RecalculateAttackPosition())
+                    {
+
+                    }
+                    else
+                    {
+                        KeepFocussingTarget(); // attack target
+                    }
                 }
                 else
                 {
                     WalkToDestination(); // walk to destination (or target)
                 }
             }
-
             Replication.Update();
         }
-
+        
         public override void OnCollision(GameObject collider)
         {
-            if (collider == null || collider == TargetUnit) // If we're colliding with the target, don't do anything.
+            /*if (collider == null || collider == TargetUnit) // If we're colliding with the target, don't do anything.
```

Comment:
Remove unused code instead of commenting out. If there is any work to be done on this method add TODO instead.